File: eps/src/frontend/src/services/daily-report.service.js
import axios from 'axios';

const API_URL = '/api/v1/daily-report';

class DailyReportService {
    submit(dailyreport) {
        return axios
            .post(API_URL + '/submit', {
                studentId: dailyreport.studentId,
                email: dailyreport.email,
                date: dailyreport.date,

                physicalCondition: dailyreport.physicalCondition,
                symptoms: dailyreport.symptoms,
                temperature: dailyreport.temperature,
                inChinaOrNot: dailyreport.inChinaOrNot,
                inShanghaiOrNot: dailyreport.inShanghaiOrNot,
                campusesToEnter: dailyreport.campusesToEnter,
                stayInCampusOrNot: dailyreport.stayInCampusOrNot,
                campusToStay: dailyreport.campusToStay,
                enterSchoolOrNot: dailyreport.enterSchoolOrNot,
                province: dailyreport.province,
                city: dailyreport.city,
                county: dailyreport.county,
                country: dailyreport.country,
                address: dailyreport.address,
                isHomeAddress: dailyreport.isHomeAddress
            })
            .then(response => {
                console.log(response)
            })
    }
    getAll() {
        return axios
            .get(API_URL)
            .then(response => {
                console.log(response)
                return response;
            })
    }
    getAllByStudentId(studentId) {
        return axios
            .get(API_URL + "/" + studentId, {
                studentId: studentId
            })
            .then(response => {
                console.log(response)
                return response;
            })
    }
    getByDate(studentId, date) {
        console.log(studentId)
        console.log(date)
        return axios
            .get(API_URL + "/" + studentId + "/" + date)
            .then(response => {
                return response;
            })
    }
}

export default new DailyReportService();

File: eps/src/main/java/com/company/dailyreport/DailyReportService.java
package com.company.dailyreport;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.util.List;

@Service
public class DailyReportService {

    private final DailyReportRepository dailyReportRepository;

    @Autowired
    public DailyReportService(DailyReportRepository dailyReportRepository) {
        this.dailyReportRepository = dailyReportRepository;
    }

    public List<DailyReport> getDailyReports() {
        return dailyReportRepository.findAll();
    }

    public List<DailyReport> getDailyReportsByStudentId(Integer studentId) {
        return dailyReportRepository.findAllByStudentId(studentId);
    }

    public DailyReport getDailyReportByDate(LocalDate date) {
        return dailyReportRepository.findByDate(date);
    }

    public void addNewDailyReport(DailyReport dailyReport) {
        var newDailyReport = DailyReport.builder()
                        .studentId(dailyReport.getStudentId())
                        .email(dailyReport.getEmail())
                        .date(dailyReport.getDate())
                        .physicalCondition(dailyReport.getPhysicalCondition())
                        .symptoms(dailyReport.getSymptoms())
                        .temperature(dailyReport.getTemperature())
                        .inChinaOrNot(dailyReport.getInChinaOrNot())
                        .inShanghaiOrNot(dailyReport.getInShanghaiOrNot())
                        .campusesToEnter(dailyReport.getCampusesToEnter())
                        .stayInCampusOrNot(dailyReport.getStayInCampusOrNot())
                        .campusToStay(dailyReport.getCampusToStay())
                        .enterSchoolOrNot(dailyReport.getEnterSchoolOrNot())
                        .province(dailyReport.getProvince())
                        .city(dailyReport.getCity())
                        .county(dailyReport.getCounty())
                        .country(dailyReport.getCountry())
                        .address(dailyReport.getAddress())
                        .isHomeAddress(dailyReport.getIsHomeAddress())
                        .status(1)
                        .build();

        dailyReportRepository.save(newDailyReport);
    }
}
